cylc/flow/async_util.py

```diff
@@ -478,3 +479,48 @@ async def _fcn(*args, executor=None, **kwargs):
 
 
 async_listdir = make_async(os.listdir)
+
+
+@asynccontextmanager
+async def async_block():
```

Comment:
Crude but effective way to allow us to call async -> sync -> async.

This makes a list of async tasks before the plugin is called in order to await any new tasks created by the plugin.

This means that plugins can go async like so:

```python
def plugin(opts):
    asyncio.create_task(run(opts))
    # the plugin will return now, but Cylc will await the "run" for us

async def run(opts):
    ...
```

I think it's a reasonable solution.

Sadly, directly converting the plugin interfaces to `async` isn't an easy option due to the use of plugins within parsec. Making the plugin async would make loading the global config async which would make global config value retrieval (e.g. the line below) async:

```python
glbl_cfg().get(['install', 'symlink dirs'])
```

Which would basically require the entire cylc-flow codebase to be async and prohibit config caching at the module level.